src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -375,389 +299,12 @@ private SuggestedActionSet InlineActions(SuggestedActionSet actionSet)
-                    // to get the span and category for the new top level suggested action.
-                    var (span, category) = CombineSpansAndCategory(suppressionSets);
-                    var wrappingSet = new SuggestedActionSet(
-                        category,
-                        actions: SpecializedCollections.SingletonEnumerable(wrappingSuggestedAction),
-                        title: CodeFixesResources.Suppress_or_Configure_issues,
-                        priority: SuggestedActionSetPriority.None,
-                        applicableToSpan: span);
-                    sets = sets.Add(wrappingSet);
-                }
-
-                return sets;
-
-                // Local functions
-                static (Span? span, string category) CombineSpansAndCategory(IEnumerable<SuggestedActionSet> sets)
-                {
-                    // We are combining the spans and categories of the given set of suggested action sets
-                    // to generate a result span containing the spans of individual suggested action sets and
-                    // a result category which is the maximum severity category amongst the set
-                    var minStart = -1;
-                    var maxEnd = -1;
-                    var category = PredefinedSuggestedActionCategoryNames.CodeFix;
-
-                    foreach (var set in sets)
-                    {
-                        if (set.ApplicableToSpan.HasValue)
-                        {
-                            var currentStart = set.ApplicableToSpan.Value.Start;
-                            var currentEnd = set.ApplicableToSpan.Value.End;
-
-                            if (minStart == -1 || currentStart < minStart)
-                            {
-                                minStart = currentStart;
-                            }
-
-                            if (maxEnd == -1 || currentEnd > maxEnd)
-                            {
-                                maxEnd = currentEnd;
-                            }
-                        }
-
-                        Debug.Assert(set.CategoryName == PredefinedSuggestedActionCategoryNames.CodeFix ||
-                                     set.CategoryName == PredefinedSuggestedActionCategoryNames.ErrorFix);
-
-                        // If this set contains an error fix, then change the result category to ErrorFix
-                        if (set.CategoryName == PredefinedSuggestedActionCategoryNames.ErrorFix)
-                        {
-                            category = PredefinedSuggestedActionCategoryNames.ErrorFix;
-                        }
-                    }
-
-                    var combinedSpan = minStart >= 0 ? new Span(minStart, maxEnd) : (Span?)null;
-                    return (combinedSpan, category);
+                    return UnifiedSuggestedActionsSource.GetFilterAndOrderCodeFixes_MustBeCalledFromUIThread(
```

Comment:
pity.  but i understand why it's necessary. hopefully we can get rid of that UI dep soon.